Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,469 @@
+    {
+
+        #region Declarations
+        /// <summary>
+        /// Ticks collection
+        /// </summary>
+        protected List<Tick> ticks = new List<Tick>();
+        CancellationTokenSource _tickerToken;
+        /// <summary>
+        /// Divisor for prices. Scales prices/volumes to allow trades on 0.01 of unit
+        /// </summary>
+        protected decimal divisor = 100;
+        readonly object _fillLock = new object();
+        const string buy = "buy";
+        const string sell = "sell";
+        /// <summary>
+        /// Currently limited to BTCUSD
+        /// </summary>
+        protected Symbol symbol = Symbol.Create("BTCUSD", SecurityType.Forex, Market.Bitfinex);
+        /// <summary>
+        /// List of known orders
+        /// </summary>
+        public ConcurrentDictionary<int, Order> CachedOrderIDs = new ConcurrentDictionary<int, Order>();
+        /// <summary>
+        /// List of filled orders
+        /// </summary>
+        protected readonly FixedSizeHashQueue<int> filledOrderIDs = new FixedSizeHashQueue<int>(10000);
+        /// <summary>
+        /// List of unknown orders
+        /// </summary>
+        protected readonly FixedSizeHashQueue<int> unknownOrderIDs = new FixedSizeHashQueue<int>(1000);
+        /// <summary>
+        /// Name of wallet
+        /// </summary>
+        protected string wallet;
+        const string _exchange = "bitfinex";
+        /// <summary>
+        /// Api Key
+        /// </summary>
+        protected string apiKey;
+        /// <summary>
+        /// Api Secret
+        /// </summary>
+        protected string apiSecret;
+        #endregion
+
+        public TradingApi.Bitfinex.BitfinexApi RestClient { get; set; }
+
+
+        /// <summary>
+        /// Create bitfinex brokerage
+        /// </summary>
+        public BitfinexBrokerage()
+            : base("bitfinex")
+        {
+            this.Initialize();
+        }
+
+        private void Initialize()
+        {
+
+            //todo: also stored in BrokerageData
+            apiSecret = Config.Get("bitfinex-api-secret");
+            apiKey = Config.Get("bitfinex-api-key");
+
+            wallet = Config.Get("bitfinex-wallet", "exchange");
+
+            if (string.IsNullOrEmpty(apiSecret))
+                throw new Exception("Missing ApiSecret in config.json");
+
+            if (string.IsNullOrEmpty(apiKey))
+                throw new Exception("Missing ApiKey in config.json");
+
+            RestClient = new BitfinexApi(apiSecret, apiKey);
+
+        }
+
+        /// <summary>
+        /// Determines if ticker polling is active
+        /// </summary>
+        public override bool IsConnected
+        {
+            get { return this._tickerToken != null && !this._tickerToken.IsCancellationRequested; }
+        }
+
+        private decimal GetPrice(Order order)
+        {
+            if (order is StopMarketOrder)
+            {
+                return ((StopMarketOrder)order).StopPrice * divisor;
```

Comment:
Great news! Bugs related to the divisor are basically a nightmare scenario, so a decimal quantity would be very welcome.